src/modules/account/RadixAccount.ts

```diff
@@ -169,7 +169,12 @@ export default class RadixAccount {
      * @returns A promise with the result of the unsubscription call
      */
     public closeNodeConnection = async () => {
-        return this.nodeConnection.unsubscribe(this.getAddress())
+        this.connectionStatus.next('DISCONNECTED')
+
+        if (this.nodeConnection) {
+            this.nodeConnection.removeListener('closed', this._onConnectionClosed)
+            return this.nodeConnection.unsubscribe(this.getAddress())
+        }
```

Comment:
I'm not that used to JavaScript magic, but what will happen if `this.nodeConnection` doesn't evaluate to true? What will this method return?

Reply:
Actually, because this is an `async` function, I don't need to explicitly return `Promise.resolve()`, the default return will resolve the promise just as well